jwst/cube_build/src/cube_match_sky.c

```diff
@@ -0,0 +1,1441 @@
+      if(iystart !=-1 && iyend == -1){
+	iyend = ny;
+	done_search_y = 1;
+      }
+
+      // set up the values for fluxv, weightv, ifluxv, varv
+      int nxy = nx * ny;
+      if(done_search_x == 1 && done_search_y ==1 && done_search_w ==1){
+	// The search above for x,y  was a crude search - now narrow the search using the distance between
+	// the spaxel center and point cloud
+	for (int ix = ixstart; ix< ixend; ix ++){
+	  for ( int iy = iystart; iy < iyend; iy ++){
+	    double ydist = fabs(yc[iy] - coord2[k]);
+	    double xdist = fabs(xc[ix] - coord1[k]);
+	    double radius = sqrt( xdist*xdist + ydist*ydist);
+
+	    if (radius <= rois_pixel[k]){
+	      // Find the index for this in spatial plane
+ 	      int index_xy = iy* nx + ix;
+	      for (int iw = iwstart; iw< iwend; iw++){
+		int index_cube = iw*nxy + index_xy;
+
+		double d1 = xdist/cdelt1;
+		double d2 = ydist/cdelt2;
+		double dxy = (d1 * d1) + (d2 * d2);
+		double d3 = (wave[k] - zc[iw])/ zcdelt3[iw];
+		//double wdist = fabs( wave[k] - zc[iw]);
+		double d32 = d3 * d3;
+		double w = d32  +  dxy;
+		double wn = -w/(scalerad_pixel[k]/cdelt1);
+		double ww = exp(wn);
+
+		double weighted_flux =  flux[k]* ww;
+		double weighted_var = (err[k]* ww) * (err[k]*ww);
+		fluxv[index_cube] = fluxv[index_cube] + weighted_flux;
+		weightv[index_cube] = weightv[index_cube] + ww;
+		varv[index_cube] = varv[index_cube] + weighted_var;
+		ifluxv[index_cube] = ifluxv[index_cube] +1.0;
+
+	      }
+	    }
+	  } // end loop over iy
+	} // end loop over ix
+
+      } // end done_search_x, done_search_y, done_search_w
+    } // end loop over point cloud
+    
+
+    // assign output values:
+
+    *spaxel_flux = fluxv;
+    *spaxel_weight = weightv;
+    *spaxel_var = varv;
+    *spaxel_iflux = ifluxv;
+
+    return 0;
+}
+
+
+
+// Match point cloud to sky and determine the weighting to assign to each point cloud  member
+// to matched spaxel based on ROI - weighting type - msm
+
+int match_point_msm(double *xc, double *yc, double *zc,
+		    double *coord1, double *coord2, double *wave,
+		    double *flux, double *err,
+		    double *rois_pixel, double *roiw_pixel,
+		    double *weight_pixel, double *softrad_pixel,
+		    double *zcdelt3,
+		    int nx, int ny, int nwave, int ncube, int npt,
+		    double cdelt1, double cdelt2,
+		    double **spaxel_flux, double **spaxel_weight, double **spaxel_var,
+		    double **spaxel_iflux) {
+
+  /*
+return values: spaxel_flux, spaxel_weight, spaxel_var, spaxel_iflux
+*/
+    double *fluxv = NULL, *weightv=NULL, *varv=NULL ;  // vectors for spaxel 
+    double *ifluxv = NULL;  // vector for spaxel
+
+    // allocate memory to hold output 
+    if (mem_alloc(ncube, &fluxv, &weightv, &varv, &ifluxv)) return 1;
+    
+    double set_zero=0.0;
+    // Set all data to zero
+    for (int i = 0; i < ncube; i++){
+      varv[i] = set_zero;
+      fluxv[i] = set_zero;
+      ifluxv[i] = set_zero;
+      weightv[i] = set_zero;
+    }
```

Comment:
Again this can be simplified as in `cube_match_internal.c` if using `alloc_flux_arrays`.

Reply:
done